main/signers/trezor/bridge.ts

```diff
@@ -0,0 +1,205 @@
+
+  // methods to send requests from the application to a Trezor device
+  async getFeatures (params: CommonParams) {
+    return this.makeRequest(() => TrezorConnect.getFeatures(params))
+  }
+
+  async getAccountInfo (device: Device, path: string) {
+    return this.makeRequest(() => TrezorConnect.getAccountInfo({ device, path, coin: 'eth' }))
+  }
+
+  async getPublicKey (device: Device, path: string) {
+    return this.makeRequest(() => TrezorConnect.getPublicKey({ device, path }))
+  }
+
+  async getAddress (device: Device, path: string, display = false) {
+    return this.makeRequest(() => TrezorConnect.ethereumGetAddress({
+      device,
+      path,
+      showOnTrezor: display
+    })).then(result => (result.address || '').toLowerCase())
+  }
+
+  async signMessage (device: Device, path: string, message: string) {
+    return this.makeRequest(() => TrezorConnect.ethereumSignMessage({
+      device,
+      path,
+      message,
+      hex: true
+    })).then(result => result.signature)
+  }
+
+  async signTypedData (device: Device, path: string, data: any) {
+    return this.makeRequest(() => TrezorConnect.ethereumSignTypedData({
+      device,
+      path,
+      data,
+      metamask_v4_compat: true,
+    })).then(result => result.signature)
+  }
+
+  async signTypedHash (device: Device, path: string, data: any, domainSeparatorHash: string, messageHash: string) {
+    return this.makeRequest(() => TrezorConnect.ethereumSignTypedData({
+      device,
+      path,
+      data,
+      domain_separator_hash: domainSeparatorHash,
+      message_hash: messageHash,
+      metamask_v4_compat: true
+    })).then(result => result.signature)
+  }
+
+  async signTransaction (device: Device, path: string, tx: any) {
+    return this.makeRequest(() => TrezorConnect.ethereumSignTransaction({
+      device,
+      path,
+      transaction: tx
+    })).then(result => {
+      const { v, r, s } = result
+      return { v, r, s }
+    })
+  }
+
+  async pinEntered (deviceId: string, pin: string) {
+    log.debug('pin entered for device', deviceId)
+
+    TrezorConnect.uiResponse({ type: UI.RECEIVE_PIN, payload: pin })
+
+    this.emit('trezor:entered:pin', deviceId)
+  }
+
+  async passphraseEntered (deviceId: string, phrase: string) {
+    log.debug('passphrase entered for device', deviceId)
+
+    TrezorConnect.uiResponse({ type: UI.RECEIVE_PASSPHRASE, payload: { save: true, value: phrase } })
+
+    this.emit('trezor:entered:passphrase', deviceId)
+  }
+
+  async enterPassphraseOnDevice (deviceId: string) {
+    log.debug('requested to enter passphrase on device', deviceId)
+
+    TrezorConnect.uiResponse({
+      type: UI.RECEIVE_PASSPHRASE,
+      payload: { value: '', passphraseOnDevice: true, save: true }
+    })
+
+    this.emit('trezor:enteringPhrase', deviceId)
+  }
+
+  private async makeRequest <T> (fn: () => Response<T>, retries = 20) {
+    try {
+      const result = await handleResponse(fn())
+      return result
+    } catch (e: any) {
```

Comment:
Error typing?

Reply:
types for variables in a `catch` clause can only be `unknown` or `any`

https://typescript.tv/best-practices/error-ts1196-catch-clause-variable-type-annotation/

Reply:
Got it, I prefer `unknown` as I aim to remove all instance of `any` from my TS